CRM/Core/BAO/Dashboard.php

```diff
@@ -233,9 +233,9 @@ public static function addDashlet(&$params) {
       $dashlet->domain_id = $params['domain_id'] ?? CRM_Core_Config::domainID();
 
       // Try and find an existing dashlet - it will be updated if found.
-      if (!empty($params['name']) || !empty($params['url'])) {
+      if (!empty($params['name']) || !empty($params['label'])) {
         $dashlet->name = $params['name'] ?? NULL;
-        $dashlet->url = $params['url'] ?? NULL;
+        $dashlet->url = $params['label'] ?? NULL;
         $dashlet->find(TRUE);
```

Comment:
I understand the overall issue, but am wondering if this is a typo: setting $dashlet->url to equal $params['label'] ?

Reply:
Thanks for pointing this, was a typo on my end :/ Updated the patch now.

Reply:
Why would we search for existing dashlets by label? Labels are not guaranteed to be unique.

The existing code has this legacy quirk where even if you don't pass an id it will still attempt to update based on `name` or `url`. Removing the matching on `url` feels like a good step forward. It removes a legacy behavior which would be nice to get rid of completely. But then this PR adds `label` in, which feels like a step backwards toward more legacy behavior. Can we just do name matching only?

Reply:
Hmm agree, updated the patch. Can you please check now?